torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -7513,3 +7513,39 @@ def test_ddp_bwd_with_retain_graph_static_single_fwd(self):
                 [param.grad for param in model_local.parameters()]
             )
             self.assertEqual(dist_grad_tensor, local_grad_tensor)
+
+        @unittest.skipIf(
+            BACKEND != "nccl" and BACKEND != "gloo",
+            "MPI backend does not support DDP communication hook on CUDA devices",
+        )
+        @skip_if_lt_x_gpu(2)
+        @skip_if_rocm
+        def test_hook_grad_idx_to_bucket_mapping(self):
+            # Test hook to ensure get_grad_index_to_variable_mapping works appropriately.
+            def _test_hook(hook_state, bucket: dist.GradBucket) -> torch.futures.Future:
+                # Run allreduce and wait for it to complete
+                allreduce_fut = default._allreduce_fut(None, bucket.get_tensor())
+                allreduce_fut.wait()
+                per_param_grad_tensors = bucket.get_per_parameter_tensors()
```

Comment:
Would I be correct if I assume these tensors are views on the bucket?

Reply:
Yep, that is correct. There's no copying/cloning of tensors going on here.